r/configure

```diff
@@ -173,6 +188,10 @@ else
         BUNDLED_LIBS=`echo "$BUNDLED_LIBS" | sed -e "s/\\.a lib/ -l/g" | sed -e "s/\\.a$//" | sed -e "s/^lib/-l/" | tr '\n' ' ' | sed -e "s/ $//"`
         PKG_DIRS="-L`pwd`/$LIB_DIR"
 
+        if [ "$ARROW_DEPENDENCY_SOURCE" == "AUTO" ]; then
+          PKG_LIBS = "pkgconfig ${LIB_DIR} $PKG_LIBS"
```

Comment:
I think you still need https://github.com/apache/arrow/pull/10710#discussion_r678380437 here, and also adding a comment explaining what's up would help our future selves.